lib/provider/chromium.ts

```diff
@@ -0,0 +1,295 @@
+        'chromium-browser-snapshots/o?delimiter=/&prefix=';
+      let os = '';
+      if (this.osType === 'Windows_NT') {
+        os = 'Win';
+        if (this.osArch === 'x64') {
+          os = 'Win_x64';
+        }
+      } else if (this.osType === 'Linux') {
+        os = 'Linux';
+        if (this.osArch === 'x64') {
+          os = 'Linux_x64';
+        }
+      } else {
+        os = 'Mac';
+      }
+      revisionUrl += os + '/';
+      let chromiumBasePosition: number = downloadJson['chromium_base_position'];
+      
+      const httpOptions = { fileName, ignoreSSL: this.ignoreSSL,
+        proxy: this.proxy };
+      while(chromiumBasePosition > 0) {
+        const revisionBasePositionUrl =
+          `${revisionUrl}${chromiumBasePosition}/`;
+        const body = await requestBody(revisionBasePositionUrl, httpOptions);
+        const jsonBody = JSON.parse(body);
+        if (jsonBody['items']) {
+          fs.writeFileSync(fileName, JSON.stringify(jsonBody, null, 2));
+          return jsonBody;
+        } else {
+          chromiumBasePosition--;
+        }
+      }
+      return null;
+    }
+  }
+
+  /**
+   * Step 4: Get the download url for the chromium zip. Unzipping the zip file
+   * directory. The folders and binaries uncompressed are different for each OS.
+   * The following is examples of each OS:
+   * 
+   * downloads/
+   *  |- chrome-linux/chrome
+   *  |- chrome-mac/Chromium.app
+   *  |- chrome-win/chrome.exe
+   * 
+   * @param storageObject The download-storage-json object
+   * @param majorVersion The major version, this must be a whole number.
+   */
+  async downloadUrl(storageObject: JsonObject, majorVersion: string
+      ): Promise<void> {
+    const fileName = path.resolve(this.outDir, 
+      this.compressedBinaryFileName.replace('.zip', `-${majorVersion}.zip`));
+    if (isExpired(fileName)) {
+      const httpOptions = { fileName, ignoreSSL: this.ignoreSSL,
+        proxy: this.proxy };
+      for (let item of storageObject['items'] as JsonObject[]) {
+        const name: string = item['name'];
+        if (name.indexOf('chrome') >= 0) {
+          const downloadUrl = item['mediaLink'];
+          await requestBinary(downloadUrl, httpOptions);
+          break;
+        }
+      }
+    }
+    unzipFile(fileName, this.outDir);
+  }
+
+  async updateBinary(majorVersion?: string): Promise<void> {    
+    const allJson = await this.downloadAllJson();
+    const downloadVersionJson = await this.downloadVersionJson(
+      allJson, majorVersion);
+    const storageObject = await this.downloadStorageObject(
+      downloadVersionJson, majorVersion);
+    await this.downloadUrl(storageObject, majorVersion);
+  }
+
+  getBinaryPath(version?: string): string | null {
+    try {
+      const configFilePath = path.resolve(this.outDir, this.configFileName);
+      return getBinaryPathFromConfig(configFilePath, version);
+    } catch (_) {
+      return null;
```

Comment:
What does it mean if there's an exception here? Maybe at least log an error?

Reply:
That's probably a good idea. This is a pattern used for all providers so I'll create a new bug to work on this.

Reply:
https://github.com/angular/webdriver-manager/issues/392